### DISCUSSION TITLE
Can I solve these coupled differential equations in MOOSE without creating a custom kernel?

### DISCUSSION THREAD

**Discussion by K-Komal-98** [idaholab/moose#24059] [Q&A Tools] [answered]
I have solved a single second order differential equation using the ADMatdiffusion and Reaction kernel. Can you assist me on how to solve this problem with the existing moose kernels?
![coupled diff eqn](https://user-images.githubusercontent.com/113084320/231685063-63bbfdee-fc49-461e-9e09-784d63f78038.png)

**Comment by GiudGiud** [comment]
Hello

The BCs are Dirichlet and Neumann, you can use off-the-shelf MOOSE BCs for this. 

You ll want to write a new kernels (start from ADCoupledForce) for the u^2, v^2 and u^2 v^2 terms.

The second order space derivatives you can use a Diffusion kernel for the one that is in its own equation.
For the other one you ll need to create a coupledDiffusion kernel, that has a "variable" argument that is different than the variable that gets the second derivative taken from.

Guillaume

**Comment by K-Komal-98** [comment, marked as answer]
Can I use ADMatDiffusion kernel for the second order space derivatives. I have written a simple code for a problem attached here. Can you please guide me if this is a correct usage or not?
![493D832B-014D-45EA-B0EB-C7A3DB183D5C](https://user-images.githubusercontent.com/113084320/233100610-c2776159-6085-4aa0-9cbb-785572c90726.png)
```
[Mesh]
    type = GeneratedMesh
    dim = 1
    nx = 60 
    xmax = 30
[]
[Variables]
    [u]
    []
    [v]
    []
[]

[Kernels]
  [1_diff_u]
    type = ADMatDiffusion
    diffusivity = 1
    variable = u
  []
  [1_diff_v]
    type = ADMatDiffusion
    diffusivity = 2
    variable = u
    v = v
  []
  [2_diff_u]
    type = ADMatDiffusion
    diffusivity = 3
    variable = v
    v = u
  []
  [2_diff_v]
    type = ADMatDiffusion
    diffusivity = 4
    variable = v
  []
[]

[BCs]
  [left_u]
    type = DirichletBC
    variable = u
    boundary = left
    value = 0
  []
  [right_u]
    type = DirichletBC
    variable = u
    boundary = right
    value = 10
  []
  [left_v]
    type = DirichletBC
    variable = v
    boundary = left
    value = 0
  []
  [right_v]
    type = DirichletBC
    variable = v
    boundary = right
    value = 10
  []
[]
[Executioner]
     type = Steady
     nl_max_its = 100
     solve_type = Newton
[]
[VectorPostprocessors]
  [u]
    type = NodalValueSampler
    sort_by = id
    variable = u
  []
  [v]
    type = NodalValueSampler
    sort_by = id
    variable = v
  []
[]
[Outputs]
  [out]
    type = CSV
    execute_on = final
  []
[]
```

**Reply by GiudGiud** [reply]
Yes this seems fine actually

**Comment by K-Komal-98** [comment]
Thank you
